features/springframework-security/src/main/java/org/opennms/web/springframework/security/OpenNMSLoginModule.java

```diff
@@ -0,0 +1,158 @@
+package org.opennms.web.springframework.security;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.FailedLoginException;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import org.opennms.netmgt.config.api.UserConfig;
+import org.opennms.netmgt.config.users.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.GrantedAuthority;
+
+public class OpenNMSLoginModule implements LoginModule {
+    private static final transient Logger LOG = LoggerFactory.getLogger(OpenNMSLoginModule.class);
+
+    private static transient volatile UserConfig m_userConfig;
+    private static transient volatile SpringSecurityUserDao m_springSecurityUserDao;
+
+    protected Subject m_subject;
+    protected CallbackHandler m_callbackHandler;
+    protected Map<String, ? super Object> m_sharedState;
+    protected Map<String, ? super Object> m_options;
+
+    protected String m_user;
+    protected Set<Principal> m_principals = new HashSet<Principal>();
+    private boolean m_debug;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void initialize(final Subject subject, final CallbackHandler callbackHandler, final Map<String, ?> sharedState, final Map<String, ?> options) {
+        LOG.info("OpenNMS Login Module initializing.");
+        for (final Entry<String, ?> entry : options.entrySet()) {
+            LOG.debug("{} = {}", entry.getKey(), entry.getValue());
+        }
+        m_subject = subject;
+        m_callbackHandler = callbackHandler;
+        m_sharedState = (Map<String, ? super Object>) sharedState;
+        m_options = (Map<String, ? super Object>) options;
+        m_debug = Boolean.parseBoolean((String) options.get("debug"));
+    }
+
+    @Override
+    public boolean login() throws LoginException {
+        LOG.debug("OpenNMSLoginModule: login()");
```

Comment:
A little, but a surprising amount of this code is not shared.  :)  Since the Karaf version has to extend the abstract karaf login module, but the other one needs to implement it fully itself.  I've pushed an update that fixes the unit tests and attempts to normalize the code as much as possible, but it's still pretty gross.